### PR TITLE
major changes dealing with configs files

### PR DESCRIPTION
**Pull request by ypriverol**

### CODE REVIEW

**Comment by husensofteng** [comment]
add `--version` (-v) to print the version of the installed tool.

File: pypgatk/ensembl/ensembl.py [hunk 1]
```diff
@@ -45,120 +45,56 @@ def __init__(self, config_file, pipeline_arguments):
...
                                              pipeline_arguments)
 
+    self._proteindb_output = 'peptide-database.fa'
     self._proteindb_output = self.get_default_parameters()[self.CONFIG_KEY_DATA][self.PROTEIN_DB_OUTPUT]
     if self.PROTEIN_DB_OUTPUT in self.get_pipeline_parameters():
       self._proteindb_output = self.get_pipeline_parameters()[self.PROTEIN_DB_OUTPUT]
+    elif self.CONFIG_KEY_DATA in self.get_default_parameters() and self.PROTEIN_DB_OUTPUT in self.get_default_parameters()[self.CONFIG_KEY_DATA]:
+      self._proteindb_output = self.get_default_parameters()[self.CONFIG_KEY_DATA][self.PROTEIN_DB_OUTPUT]
 
-    self._translation_table = self.get_default_parameters()[self.CONFIG_KEY_DATA][self.TRANSLATION_TABLE]
```
Comment (husensofteng) [inline pypgatk/ensembl/ensembl.py:52]: > File "py-pgatk/pypgatk/ensembl/ensembl.py", line 49, in __init__
>     self._proteindb_output = self.get_default_parameters()[self.CONFIG_KEY_DATA][self.PROTEIN_DB_OUTPUT]
> KeyError: 'ensembl_translation'

command: 
`!pypgatk_cli.py dnaseq-to-proteindb --input_fasta ensembl_files/Meleagris_gallopavo.Turkey_5.1.106.fa --output_proteindb lncRNAs.fa --skip_including_all_cds --include_biotypes lncRNA`

File: pypgatk/ensembl/ensembl.py [hunk 2]
```diff
@@ -45,120 +45,56 @@ def __init__(self, config_file, pipeline_arguments):
...
+    value_return = default_value
+    if variable in self.get_pipeline_parameters():
+      value_return = self.get_pipeline_parameters()[variable]
+    elif self.CONFIG_KEY_DATA in self.get_default_parameters() and \
+            self.CONFIG_KEY_VCF in self.get_default_parameters()[self.CONFIG_KEY_DATA] and \
+            variable in self.get_default_parameters()[self.CONFIG_KEY_DATA][self.CONFIG_KEY_VCF]:
+      value_return = self.get_default_parameters()[self.CONFIG_KEY_DATA][self.CONFIG_KEY_VCF][variable]
+    return value_return
 
   def three_frame_translation(self, input_fasta):
```
Comment (husensofteng) [inline pypgatk/ensembl/ensembl.py]: @ypriverol, it seems the parameter values are not read from the config file
e.g. the following generates a list of ~5k proteins from lncRNA transcripts:
`pypgatk_cli.py dnaseq-to-proteindb --config_file config/ensembl_config.yaml --input_fasta ensembl_files/Meleagris_gallopavo.Turkey_5.1.106.fa --output_proteindb lncRNAs.fa --skip_including_all_cds  --include_biotypes lncRNA`

however, when the `--include_biotypes` param is omitted then the output is empty although it has been set in the config file

File: pypgatk/toolbox/general.py [hunk 3]
```diff
@@ -171,185 +185,194 @@ def download_file(file_url: str, file_name: str, log: logging, url_file = None)
...
+
+    if url_file is not None:
+        url_file.write("{}\t{}\n".format(file_url, file_name))
+        return file_name
+
+    remaining_download_tries = REMAINING_DOWNLOAD_TRIES
+    downloaded_file = None
+    while remaining_download_tries > 0:
+        try:
+            downloaded_file, error_code = request.urlretrieve(file_url, file_name)
```
[the remark this answers is not in the document]
Comment (ypriverol) [reply pypgatk/toolbox/general.py:202]: @sonatype-lift ignore

**Comment by ypriverol** [comment]
@sonatype-lift ignore

**Review by husensofteng — APPROVED** [review]
All changes are tested (manual and automated tests).

**Pull request merged**